keps/sig-node/127-usernamespaces-support/README.md

```diff
@@ -0,0 +1,1122 @@
+owned by a user different than the effective user of that process in a volume
+that doesn't support the semantics of `fsGroup` (doesn't support
+[`SetVolumeOwnership`](https://github.com/kubernetes/kubernetes/blob/00da04ba23d755d02a78d5021996489ace96aa4d/pkg/volume/volume_linux.go#L42)
+that updates permissions and ownership of the files to be accesible by the
+`fsGroup` group ID). Such pods should be run in `Host` mode.
+
+#### Container Runtime Support
+
+- **Docker**:
+  Docker only supports a [single ID
+  mappings](https://docs.docker.com/engine/security/userns-remap/) shared by all
+  containers running in the host. There is not support for [multiple ID
+  mappings](https://github.com/moby/moby/issues/28593) yet. Updating dockershim
+  is out of scope of this KEP as [it's going to be
+  deprecated]((https://github.com/kubernetes/enhancements/pull/1985/)) and it
+  offers a very limited support for user namespaces.
+- **containerd**:
+  It's quite straigtforward to implement the CRI changes proposed below in
+  containerd/cri, we did it in
+  [this](https://github.com/kinvolk/containerd-cri/commits/mauricio/userns_poc)
+  PoC.
+- **cri-o**:
+CRI-O recently [added](https://github.com/cri-o/cri-o/pull/3944) support for
+  user namespaces through a pod annotation. The extensions to make it work with
+  the CRI changes proposed here are small.
+- gVisor, katacontainers: It's still to investigate.
+
+containerd and cri-o will provide support for the 3 possible values of `userNamespaceMode`.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+#### Breaking Existing Workloads
+
+Some features that don't work when the host user namespace is not shared are:
+
+- **Some Capabilities**:
+  The Linux kernel takes into consideration the user namespace a process is
+  running in while performing the capabilities check. There are some
+  capabilities that are only available in the initial (host) user namespace such as
+  `CAP_SYS_TIME`, `CAP_SYS_MODULE` & `CAP_MKNOD`.
+
+  If a pod is given one of those capabilities it will still be deployed, but the
+  capability will be ineffective and processes using those capabilities will
+  fail. This is not impacting the implementation in Kubernetes. If users need
+  the capability to be effective, they should use `userNamespaceMode=Host`.
+
+  The list of such capabilities is likely to change from one Linux version to
+  another. For example, Linux now has [time
+  namespaces](https://man7.org/linux/man-pages/man7/time_namespaces.7.html) and
+  there are ways to make `CAP_SYS_TIME` work inside a user namespace. There are
+  also discussions to make `CAP_MKNOD` work in user namespaces.
```

Comment:
If this KEP is implemented as propossed, I assume CAP_SYS_TIME will just work withon any need to patch Kubernetes/containerd, right? Just checking :)